### PR TITLE
[DOCS] URI parts processor

### PR DESCRIPTION
**Pull request by danhermann**
Adds documentation for the ingest processor added in https://github.com/elastic/elasticsearch/pull/65150.

### CODE REVIEW

File: docs/reference/ingest/processors/uri-parts.asciidoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+[[uri-parts-processor]]
+=== URI parts processor
+++++
+<titleabbrev>URI Parts</titleabbrev>
+++++
+
+The URI parts processor parses a text field containing a Uniform Resource
+Identifier (URI) and writes its components to the specified field.
+Components that may be extracted from the URI include domain, path, fragment,
+port, query, scheme, user info, username, and password.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/uri-parts.asciidoc:12]: I tried to streamline this a little bit. If you decide to keep the original text, I'd remove `text field` so users don't confuse this with our `text` field type.

```suggestion
Parses a Uniform Resource Identifier (URI) string and extracts its components as
an object. This URI object includes properties for the URI's domain, path, fragment,
port, query, scheme, user info, username, and password.
```

File: docs/reference/ingest/processors/uri-parts.asciidoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+[[uri-parts-options]]
+.URI Parts Options
+[options="header"]
+|======
+| Name               | Required  | Default | Description
+| `field`            | yes       | -       | The field in the input document
+containing text that may be parsed as a URI.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/uri-parts.asciidoc:20]:
```suggestion
| `field`            | yes       | -       | Field containing the URI string.
```

File: docs/reference/ingest/processors/uri-parts.asciidoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+| `target_field`     | no        | `url`   | Field under which the components
+of the parsed URI should be written.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/uri-parts.asciidoc:22]:
```suggestion
| `target_field`     | no        | `url`   | Output field for the URI object.
```

File: docs/reference/ingest/processors/uri-parts.asciidoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+| `keep_original`    | no        | true    | If true, the unparsed contents of
+`field` will be copied to `<target_field>.original`.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/uri-parts.asciidoc:24]:
```suggestion
| `keep_original`    | no        | true    | If `true`, the processor copies the
unparsed URI to `<target_field>.original`.
```

File: docs/reference/ingest/processors/uri-parts.asciidoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+| `remove_if_successful` | no    | false   | If `true` and the processor
+successfully parses the contents of `field` as a URI, then `field` will be
+removed from the document.
```
Comment (jrodewig) [inline docs/reference/ingest/processors/uri-parts.asciidoc:27]:
```suggestion
| `remove_if_successful` | no    | false   | If `true`, the processor removes
the `field` after parsing the URI string. If parsing fails, the processor does not
remove the `field`.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left a few suggestions for clarity but nothing I'd consider blocking. Thanks @danhermann!

**Comment by danhermann** [comment]
@jrodewig, thanks for the review and the suggestions. They definitely improved my rambling circumlocutions. 😃

**Pull request merged**